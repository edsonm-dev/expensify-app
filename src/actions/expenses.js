import {v4 as uuidv4} from 'uuid'

export const addExpense = (
    {
        description='',
        note='',
        amount=0,
        createdAt=0
    }={}
    )=>{
    return {
        type:'ADD_EXPENSE',
        expense:{
        id:uuidv4(),
        description,
        note,
        amount,
        createdAt  
        } 
    }
}
export const removeExpense = (
    {id=''}={}
    )=>{
    return {
        type:'REMOVE_EXPENSE',
        id  
    }
}

export const editExpense=({id='',updates={}}={}) =>{
    console.log({
        type:'EDIT_EXPENSE',
        id,
        updates
    });
    
    return {
        type:'EDIT_EXPENSE',
        id,
        updates
    }
}