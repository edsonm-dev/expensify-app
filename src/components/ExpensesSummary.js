import React from 'react'
import {connect} from 'react-redux'
import getExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'
import {Link} from 'react-router-dom'





export const  ExpensesSummary=(props)=>{


    const expensesCount = props.expenses.length
    const expensesTotal = getExpensesTotal(props.expenses)
    const expenseWord = expensesCount ===1? 'expense':'expenses'
    
    return(
    <div className="page-header">
        <div className="content-container">
        {expensesCount===0? (<p></p>): <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{numeral(expensesTotal).format('$0,0.00')}</span></h1>}
        <div className="page-header__actions">
            <Link className="button" to="/create">Add Expense</Link>
        </div>
        </div>
    </div>
    )




}






const mapStateToProps =(state)=>{
    return{
        expenses:selectExpenses(state.expenses,state.filters)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)