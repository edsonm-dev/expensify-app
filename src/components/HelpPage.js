import React from 'react';
import ReactDOM from 'react-dom';


const HelpPage = ()=>(
    <div>This is help page</div>
)

export default HelpPage;