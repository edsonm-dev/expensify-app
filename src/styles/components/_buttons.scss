.button {
    border:none;
    border-radius: 3px;
    background:$light-blue;
    color:white;
    font-size:$font-size-large;
    font-weight: 300;
    padding:$s-size $l-size;
    display:inline-block;
    text-decoration:none;
    line-height:1;

/* 
    &:hover{
        background:lighten($light-blue,5%);
    } */

}
.button--logout{
    background:none;
}
.button--secondary{
    background:#888888;
    
}