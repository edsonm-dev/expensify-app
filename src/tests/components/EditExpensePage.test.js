import React from 'react'
import {shallow} from 'enzyme'
import {EditExpensePage} from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'


let editExpense,startRemoveExpense, history, wrapper;

beforeEach(()=>{
    editExpense=jest.fn()
    removeExpense=jest.fn()
    history = {push:jest.fn()}
    wrapper = shallow(<EditExpensePage editExpense={editExpense} removeExpense={startRemoveExpense} history ={history} expense={expenses[1]} />)
})


test('should render editexpense',()=>{
    expect(wrapper).toMatchSnapshot()
})

test('should handle editexpense',()=>{

    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id,expenses[1])
})
test('should handle removeexpense',()=>{

    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(startRemoveExpense).toHaveBeenLastCalledWith({id:expenses[1].id})
})