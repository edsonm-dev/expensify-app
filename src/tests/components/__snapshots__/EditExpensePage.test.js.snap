// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render editexpense 1`] = `undefined`;
